presto-product-tests/src/main/java/io/prestosql/tests/hive/acid/TestPartitionedInsertOnlyAcidTable.java

```diff
@@ -0,0 +1,214 @@
+ */
+package io.prestosql.tests.hive.acid;
+
+import com.google.common.net.HostAndPort;
+import com.google.inject.Inject;
+import io.prestosql.plugin.hive.authentication.NoHiveMetastoreAuthentication;
+import io.prestosql.plugin.hive.metastore.thrift.ThriftHiveMetastoreClient;
+import io.prestosql.plugin.hive.metastore.thrift.Transport;
+import io.prestosql.tempto.ProductTest;
+import io.prestosql.tempto.Requirement;
+import io.prestosql.tempto.RequirementsProvider;
+import io.prestosql.tempto.configuration.Configuration;
+import io.prestosql.tempto.context.ThreadLocalTestContextHolder;
+import io.prestosql.tempto.fulfillment.table.MutableTablesState;
+import io.prestosql.tempto.fulfillment.table.hive.HiveTableDefinition;
+import io.prestosql.tempto.query.QueryExecutor;
+import io.prestosql.tempto.query.QueryResult;
+import org.apache.hadoop.hive.metastore.api.AllocateTableWriteIdsRequest;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.thrift.TException;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.UUID;
+
+import static io.prestosql.tempto.Requirements.compose;
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+import static io.prestosql.tempto.fulfillment.table.TableRequirements.mutableTable;
+import static io.prestosql.tempto.query.QueryExecutor.query;
+import static io.prestosql.tests.TestGroups.HIVE_ACID;
+import static io.prestosql.tests.TestGroups.PROFILE_SPECIFIC_TESTS;
+
+public class TestPartitionedInsertOnlyAcidTable
+        extends ProductTest
+        implements RequirementsProvider
+{
+    private static String tableName = "ORC_acid_single_int_column_partitioned";
+
+    private static final HiveTableDefinition SINGLE_INT_COLUMN_PARTITIONED_ORC = singleIntColumnPartitionedTableDefinition(Optional.empty());
+
+    private static HiveTableDefinition singleIntColumnPartitionedTableDefinition(Optional<String> serde)
+    {
+        return HiveTableDefinition.builder(tableName)
+                .setCreateTableDDLTemplate(buildSingleIntColumnPartitionedTableDDL(serde))
+                .setNoData()
+                .build();
+    }
+
+    private static String buildSingleIntColumnPartitionedTableDDL(Optional<String> rowFormat)
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("CREATE %EXTERNAL% TABLE IF NOT EXISTS %NAME%(");
+        sb.append("   col INT");
+        sb.append(") ");
+        sb.append("PARTITIONED BY (part_col INT) ");
+        if (rowFormat.isPresent()) {
+            sb.append("ROW FORMAT ").append(rowFormat.get());
+        }
+        sb.append(" STORED AS ORC");
+        sb.append(" TBLPROPERTIES ('transactional_properties'='insert_only', 'transactional'='true')");
+        return sb.toString();
+    }
+
+    @Inject
+    private MutableTablesState tablesState;
```

Comment:
I don't find this mechanism helpful. In fact, it would be easier to have a helper method which creates the table and use it in directly in every test.
Bonus advantage is that you can easily reduce boiler plate by merging partitioned and unpartitioned tests together, by parameterizeding the table creation code.

Reply:
It would reduce the code but it would increase the runtime of tests as for each test we would drop-create table. Ideal case would be to have parametrization available to @BeforeClass method to but that does not seem to be there, closest is @Factory annotation that can help here.

Reply:
Moved tests to parametrized inputs as after changes I didnt need same table across multiple tests.

Reply:
mutable test is also recreated before each test

Reply:
This file is deleted now.